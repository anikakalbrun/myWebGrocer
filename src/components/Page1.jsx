import React from 'react';
import {func} from 'prop-types';

import Button from './Button';
import Wrapper from './Wrapper';
import '../style/css/Page.css';

Page1.propTypes = {
    goToSecondPage: func.isRequired
};

export default function Page1({goToSecondPage}) {
    return (
        <Wrapper>
            <h1 className='presentation__title'>Welcome</h1>
            <p>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.
                Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.</p>
            <p>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.
                Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis,
                ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo.
                Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim.
                Donec pede justo</p>
            <Button text={'Next'}
                    btnClassName='button--white pull-right m-t-10'
                    callback={goToSecondPage}
            />
        </Wrapper>
    );
}