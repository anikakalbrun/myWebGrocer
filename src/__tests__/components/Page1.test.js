import React from 'react';
import Page1 from '../../components/Page1';

let wrapper;
const goToSecondPage = jest.fn();

describe('Page1', () => {

    beforeEach(() => {
        wrapper = shallow(<Page1 goToSecondPage={goToSecondPage}/>);
    });

    it('renders the Wrapper component', () => {
        expect(wrapper.find('Wrapper').length).toEqual(1)
    });

    it('renders proper h1', () => {
        expect(wrapper.find('Wrapper').at(0).props().children[0]).toMatchSnapshot()
    });

    it('renders the first paragraph with proper content', () => {
        expect(wrapper.find('Wrapper').at(0).props().children[1]).toMatchSnapshot()
    });

    it('renders the second paragraph with proper content', () => {
        expect(wrapper.find('Wrapper').at(0).props().children[2]).toMatchSnapshot()
    });

    it('renders the Button component and passes props to it', () => {
        expect(wrapper.find('Wrapper').at(0).props().children[3]).toMatchSnapshot()
    });

    it('button click calls setPageNumber', () => {
        wrapper.find('Wrapper').at(0).dive().find('Button').dive().find('button').simulate('click');
        expect(goToSecondPage).toHaveBeenCalled()
    });
});
