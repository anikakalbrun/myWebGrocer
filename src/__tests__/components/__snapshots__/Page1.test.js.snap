// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page1 renders proper h1 1`] = `
<h1
  className="presentation__title"
>
  Welcome
</h1>
`;

exports[`Page1 renders the Button component and passes props to it 1`] = `
<Button
  btnClassName="button--white pull-right m-t-10"
  callback={[Function]}
  disabled={false}
  text="Next"
  type="button"
/>
`;

exports[`Page1 renders the first paragraph with proper content 1`] = `
<p>
  Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
</p>
`;

exports[`Page1 renders the second paragraph with proper content 1`] = `
<p>
  Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo
</p>
`;
